src/main/java/com/idrsolutions/microservice/BaseServletContextListener.java

```diff
@@ -150,8 +151,37 @@ protected void validateConfigFileValues(final Properties propertiesFile) {
         validateIndividualTTL(propertiesFile);
         validateFileDeletionService(propertiesFile);
         validateFileDeletionServiceFrequency(propertiesFile);
+        validateFilePathLimit(propertiesFile); //Ensure this is done after input and output path are validated.
     }
 
+
+    private static void validateFilePathLimit(final Properties properties) throws IllegalArgumentException {
+        final String filePathLimit = properties.getProperty(KEY_PROPERTY_FILEPATH_LIMIT);
+        if (filePathLimit == null || filePathLimit.isEmpty() || !filePathLimit.matches("\\d+") || Integer.parseInt(filePathLimit) == 0) {
+            properties.setProperty(KEY_PROPERTY_FILEPATH_LIMIT, "250");
+            final String message = String.format("Properties value for \"filePathLimit\" was set to \"%s\" but should be a positive integer. Using a value of 250.", filePathLimit);
+            LOG.log(Level.WARNING, message);
+        }
+
+        final int iFilePathLimit = Integer.parseInt(properties.getProperty(BaseServletContextListener.KEY_PROPERTY_FILEPATH_LIMIT));
+        final int previewLength = 19;
+        final int uuidLength = 37;
+        final int outputPath = properties.getProperty(KEY_PROPERTY_OUTPUT_PATH).length() + uuidLength + previewLength;
+        final int inputPath = properties.getProperty(KEY_PROPERTY_INPUT_PATH).length() + uuidLength;
+
+        final int longestPath = Math.max(outputPath, inputPath);
+
+        final int filenameCap = iFilePathLimit - longestPath;
+        if (filenameCap < 1) {
+            final String message = "The \"filePathLimit\" must be large enough to cover the longest of the \"inputPath\"/\"outputPath\" (" + longestPath + "), uuid (37), preview output (19), and the filename (at least 1). The value must be at least " + (longestPath + 37 + 19 + 1) + ", but 250 is recommended for compatibility across platforms.";
```

Comment:
Rather than explicitly using 19/37 here, would it not be preferable to use previewLength/uuidLength?
```suggestion
            final String message = "The \"filePathLimit\" must be large enough to cover the longest of the \"inputPath\"/\"outputPath\" (" + longestPath + "), uuid (" + uuidLength + "), preview output (" + previewLength + "), and the filename (at least 1). The value must be at least " + (longestPath + uuidLength + previewLength + 1) + ", but 250 is recommended for compatibility across platforms.";
```